maven-plugin-annotations/src/main/java/org/apache/maven/plugins/annotations/Execute.java

```diff
@@ -39,20 +39,31 @@
 public @interface Execute
 {
     /**
-     * lifecycle phase to fork. Note that specifying a phase overrides specifying a goal.
+     * Lifecycle phase to fork. Note that specifying a phase overrides specifying a goal.
+     * For custom lifecycle phase ids use {@link #customPhase()} instead.
+     * This value takes precedence over the {@link #customPhase()}.
      * @return the phase
      */
     LifecyclePhase phase() default LifecyclePhase.NONE;
 
     /**
-     * goal to fork. Note that specifying a phase overrides specifying a goal. The specified <code>goal</code> must be
+     * Custom lifecycle phase to fork. Note that specifying a {@link #phase} overrides the specified custom phase.
+     * Also specifying a custom phase overrides specifying a goal.
+     * This element should only be used for non-standard phases. For standard phases rather use {@link #phase()}.
+     * 
+     * @since 3.8.0
+     */
+    String customPhase() default "";
```

Comment:
maybe an error or at least warning if user defined both
In many case users don't read a documentation.

Reply:
Agree, with https://github.com/apache/maven-plugin-tools/pull/170/commits/9ccc3223cf3e48e01668c72e15d58b9b05a25577 an exception is thrown in that case.